### PR TITLE
Add multi-select support for db view

### LINKED ISSUE #455: Allow multi-select in database view
Similar to #454, but for the databases view. Here, we want to support multi-select for:

* remove database
* upgrade database
* show database directory (possibly)

### PR DESCRIPTION
**Pull request by aeisenberg**
These commands now work on multiple databases:

* remove databases
* open external folder
* upgrade databases## Checklist

Fixes #455.

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [x] `@github/product-docs-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

**Comment by aeisenberg** [comment]
Ping @github/product-docs-dsp!

Similar to #456, but for the databases view.

**Comment by shati-patel** [comment]
Thanks for the ping! I've added this to our existing docs issue for "multi-selecting things" and we'll check if anything needs updating 😃

File: extensions/ql-vscode/src/databases-ui.ts [hunk 1]
```diff
@@ -213,113 +330,174 @@ export class DatabaseUI extends DisposableObject {
...
     }
-  }
-
-  handleChooseDatabaseInternet = async (): Promise<DatabaseItem | undefined> => {
-    return await promptImportInternetDatabase(this.databaseManager, this.storagePath);
-  }
+  };
+
+  handleChooseDatabaseInternet = async (): Promise<
+    DatabaseItem | undefined
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/databases-ui.ts:336]: I've been trying to ignore these reformattings that (to my eye) unnecessarily split lines, but splitting up this type is really weird, given that `Promise<DatabaseItem | undefined>` is kept together on the next function. Is there a setting that's setting a column limit to something significantly less than 80 columns?

File: extensions/ql-vscode/src/databases-ui.ts [hunk 2]
```diff
@@ -213,113 +330,174 @@ export class DatabaseUI extends DisposableObject {
...
-      logger.log('Received request to upgrade database, but database contents could not be found.');
-      return;
-    }
-    if (databaseItem.contents.dbSchemeUri === undefined) {
-      logger.log('Received request to upgrade database, but database has no schema.');
-      return;
-    }
+    await this.handleUpgradeDatabase(
+      this.databaseManager.currentDatabaseItem,
+      []
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/databases-ui.ts:370]: For consistency with how multi-query commands were handled, shouldn't this empty list be either `undefined` or a singleton list of `this.databaseManager.currentDatabaseItem`?

Comment (jcreedcmu) [reply extensions/ql-vscode/src/databases-ui.ts:370]: I guess based on the way it's implemented it doesn't really matter. I'm very much not a fan of this pattern of `(singleThing: T, multipleThings: T[])` where allegedly one or the other can be undefined, and usually the `multipleThings` includes the `singleThing`, but vscode's API imposes it on us at the command callback interface, so I guess it's fine.

File: extensions/ql-vscode/src/databases-ui.ts (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/databases-ui.ts:336]: This is the standard prettier reformatting (recognizing some flags in our settings). And it is aggressive about short lines. I know we're not using prettier in the build. It is currently set to 80 columns. I can make it 100 from now on.

File: extensions/ql-vscode/src/databases-ui.ts (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/databases-ui.ts:370]: I'm disappointed with how under-specified this part of the API is. There's no clear location where it's described how the multiselection passes arguments as far as I can tell. I had to figure this out trial and error.

**Pull request merged**